optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -172,9 +172,20 @@ public SupplyManager getSupplyManager() {
     // Complex methods
     // ************************************************************************
 
+    private void assertWorkingSolutionClassSameAsAnnotatedClass(Solution_ workingSolution) {
+        Class<Solution_> actualSolutionClass = (Class<Solution_>) Objects.requireNonNull(workingSolution).getClass();
+        Class<Solution_> expectedSolutionClass = getSolutionDescriptor().getSolutionClass();
+        if (!actualSolutionClass.equals(expectedSolutionClass)) {
+            throw new IllegalStateException("Working solution class (" + actualSolutionClass.getCanonicalName() +
+                    ") different from " + PlanningSolution.class + "-annotated class (" +
```

Comment:
style: other error messages output " from the @PlanningSolution annotated class",
instead of " from PlanningSolution-annotated class".